src/Core/Cart/Calculator.php

```diff
@@ -192,6 +208,19 @@ public function getDiscountTotal()
             $amount = $amount->add($cartRule->getDiscountApplied());
         }
 
+        $totalWithoutDiscount = $this->getRowTotalWithoutDiscount();
+        $initialShipping = $this->getFees()->getInitialShippingFees();
+        $finalShipping = $this->getFees()->getFinalShippingFees();
+
+        if (null !== $initialShipping && null !== $finalShipping) {
```

Comment:
I'm not sure I get this part Part of the shipping is added, the other one is subsided.. Which one is what? Maybe this justifies a comment

Reply:
Since you compute the whole discount separately, is it still useful to `applyFlatDiscount` on rows during the computation process?

Reply:
> I'm not sure I get this part Part of the shipping is added, the other one is subsided.. Which one is what? Maybe this justifies a comment

You're right, I'll add a comment explaining this part!

>Since you compute the whole discount separately, is it still useful to `applyFlatDiscount` on rows during the computation process?

I think it's not useful but I'm pretty sure that getting rid of it will introduce a BC break?

Reply:
Ok for `applyFlatDiscount`

For the shipping part I didn't get it at first but with your comment it seems logical, but I think we could simplify this an remove the comment with this:
```php
if (null !== $initialShipping && null !== $finalShipping) {
    $shippingDiscount = (new AmountImmutable())->add($initialShipping)->sub($finalShipping);
    $totalWithoutDiscount = $totalWithoutDiscount->sub($shippingDiscount);
}
```
or to avoid the intermediate variables:
```php
if (null !== $this->getFees()->getInitialShippingFees() && null !== $this->getFees()->getFinalShippingFees()) {
    $shippingDiscount = (new AmountImmutable())
                         ->add($this->getFees()->getInitialShippingFees())
                         ->sub($this->getFees()->getFinalShippingFees())
    ;
    $totalWithoutDiscount = $totalWithoutDiscount->sub($shippingDiscount);
}
```
What do you think?

Reply:
Just updated the code 😉